website/utils.py

```diff
@@ -9,22 +10,64 @@ def get_details_from_post_code(postcode: str) -> dict:
     return response
 
 
-def get_long_lat(details: dict) -> tuple:
-    """Given a dictioanry, find the longitude and latitude in that order."""
-    print('a')
+def get_postcode_long_lat(details: dict) -> tuple:
+    """Given a dictionary, find the longitude and latitude in that order."""
     return (details['result']['longitude'], details['result']['latitude'])
 
 
-def get_location_name(details: dict) -> str:
-    """Given the location details extract the location name"""
-    return details['result']['nuts']
+def get_location(address: dict) -> str:
+    """Extract the city/town/village name of a location."""
+    if 'city' in address:
+        return address['city']
+    if 'town' in address:
+        return address['town']
+    if 'village' in address:
+        return address['village']
+    return ""
 
 
-def get_county(details: dict) -> str:
-    """Given the location details extract the county."""
-    return details['result']['admin_county']
+def get_county(address: dict) -> str:
+    """Extract the county name of a location."""
+    if 'county' in address:
+        return address['county']
+    if 'state_district' in address:
+        return address['state_district']
+    return ""
```

Comment:
can condense this, `return address["county"] if "county" in address else address.get("state_district","") `